### PR TITLE
add code for alphanumeric sorting and formatting of OWNERS_ALIASES

### PR DESCRIPTION
**Pull request by neolit123**
- add hack/format-owners-aliases.go and unit tests in _test.go
- call the unit tests in hack/verify-owners-aliases-test.sh
- add hack/update-owners-aliases.sh that will update the OWNERS_ALIASES
file in place
- modify verify-generated-docs.sh and Makefile to include
the new verification
- include a couple of new dependencies in go.* files
- update the actual OWNERS_ALIASES file to remove duplicate owners
and sort the groups and owners

**Which issue(s) this PR fixes**:

NONE

/kind feature
/area test
/priority important-longterm
/assign @justaugustus

### CODE REVIEW

**Comment by neolit123** [comment]
maybe this should be in the prow "owners" plugin, but i would leave this refactor move to someone else.

File: hack/format-owners-aliases.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	i := 0
+	for k := range a.Aliases {
+		keys[i] = k
+		// remove dupes and sort the list of OWNERS
+		a.Aliases[k], _ = removeDetectDupes(a.Aliases[k], false)
+		sort.Strings(a.Aliases[k])
+		i++
+	}
+	// error out on duplicate keys (group names), otherwise sort the keys
+	keys, err := removeDetectDupes(keys, true)
```
Comment (neolit123) [inline hack/format-owners-aliases.go]: the better alternative here might be auto-merge duplicate groups:
```
A:
  - A
  - B
A:
  - A
  - B
  - C
```
to result in:
```
A:
  - A
  - B
  - C
```

Comment (neolit123) [reply hack/format-owners-aliases.go]: actually, this is too entangled because the source is a `map` with multiple keys.

**Comment by neolit123** [comment]
i realized that OWNERS_ALIASES files have comments:
https://github.com/kubernetes/enhancements/blob/master/OWNERS_ALIASES#L165-L172
`yaml.v{2|3}` does not support unmarshaling of comments, so this process will discard them.

this is a trade off.

**Comment by neolit123** [comment]
> yaml.v{2|3} does not support unmarshaling of comments, so this process will discard them.
correction, looks like yaml.v3 does support this:

updated the PR to support:
```
aliases:
  some-sig:
    - some-owner: # comment
```

but other comments e.g. on top of "some-owner" or right from "some-sig" are not supported.

**Comment by nikhita** [comment]
I think we could instead update the following part in the generator code:

https://github.com/kubernetes/community/blob/558c3e7ccc1a402a731df31ae0459d1490a16f7a/generator/app.go#L186-L194

Right now, this sorts chairs, TLs and emeritus leads separately so that they are in alphabetical order in the README. Example - https://github.com/kubernetes/community/tree/master/sig-auth#leadership

We could modify it to update such that it additionally sorts both chairs and TLs together before generating OWNERS_ALIASES:

https://github.com/kubernetes/community/blob/558c3e7ccc1a402a731df31ae0459d1490a16f7a/generator/app.go#L500-L505

**Comment by neolit123** [comment]
@nikhita 

> I think we could instead update the following part in the generator code:

this seems like a much better option.
i will send a separate PR for that and maybe close this one.

**Comment by neolit123** [comment]
closing in favor of https://github.com/kubernetes/community/pull/4436
/close

**Pull request closed without merging**